up-l3/udiscovery/v3/service.adoc

```diff
@@ -0,0 +1,253 @@
+--
+
+To replicate removal of `UServiceTopic` metadata:
+[.specitem,oft-sid="dsn~discovery-setservicetopic-remove~1",oft-needs="impl,test"]
+--
+* `ttl` value in `SetServiceTopicsRequest` *MUST* be set to 0.
+--
+
+`SetServiceTopics()` Failure Reasons:
+
+[.specitem,oft-sid="dsn~discovery-setservicetopic-error-invalid~1",oft-needs="impl,test"]
+--
+*MUST* return `UCode.INVALID_ARGUMENT` if the URI passed does not contain a valid ue_id or a FQURI.
+--
+  
+[.specitem,oft-sid="dsn~discovery-setservicetopic-error-permission~1",oft-needs="impl,test"]
+--
+* *MUST* return `UCode.PERMISSION_DENIED` if the caller is not uDiscovery service in the correct recursive order (i.e. local -> domain -> central).
+* *MAY* grant access to deployment specific uEs such as a the software manager that installs or removes uEs.
+--
+
+[.specitem,oft-sid="dsn~discovery-setservicetopic-error-notfound~1",oft-needs="impl,test"]
+--
+* *MUST* return `UCode.NOT_FOUND` if no topic matching the URI was found when the `UServiceTopic` is being removed (i.e. `ttl=0`).
+--
+
+
+
+.Replication API Design
+[#udiscovery-replication-design]
+[mermaid]
+ifdef::env-github[[source,mermaid]]
+----
+sequenceDiagram
+
+OTAClient ->> Local-UDiscovery: SetServiceTopics()
+Local-UDiscovery -->> OTAClient: SetServiceTopicsResponse
+
+Local-UDiscovery ->> Domain-UDiscovery: SetServiceTopics()
+Domain-UDiscovery -->> Local-UDiscovery: SetServiceTopicsResponse
+
+Domain-UDiscovery ->> Central-UDiscovery: SetServiceTopics()
+Central-UDiscovery -->> Domain-UDiscovery: SetServiceTopicsResponse
+----
+
+
+== Implementation Consideration
+
+It is possible the information in domain or central instances might become out of sync with what is stored in the other instances. In order to rectify this situation:
+
+[.specitem,oft-sid="dsn~discovery-data-reconciliation~1",oft-needs="impl,test"]
+--
+* Domain and central instances *MUST* ensure that the information within its database is accurate and up to date and reconcile any differences periodically using the client and replication APIs described in this specification.
+--
+
+[.specitem,oft-sid="dsn~discovery-data-reconciliation-frequency~1",oft-needs="impl,test"]
+--
+* *MUST* provide an implementation of data reconciliation using the above APIs where the frequency of reconciliation is customizable so the deployment can adjust the frequency based on the deployment needs.
+--
+
+[.specitem,oft-sid="dsn~discovery-data-reconciliation-data~1",oft-needs="impl,test"]
+--
+* When reconciling data, the data fetched *MUST* be used in lieu of current cached data. 
+--
+
+[.specitem,oft-sid="dsn~discovery-data-reconciliation-not-found~1",oft-needs="impl,test"]
+--
+* If `GetServiceTopics()` returns `UCode.NOT_FOUND` for data we had in our local instance, we *MUST* flush the local cache and then replicate the deletion. we fetch information to reconcile the data If the reconciliation fails, the service *MUST* log the error and continue to use the current cached data.
```

Comment:
I am not sure who is `we` in these sentences. Can you rephrase this being (super-)explicit about the roles?

Reply:
Cleaned up the requirements.